GDEV-201 add aws-backup module

syousry6 opened the pull request.

Commit:
add aws-backup module

Commit:
GDEV-201 fixing bracket issue

Commit:
GDEV-201 removing un used variables

Commit:
removing kms arn data

Commit:
GDEV-201 fixing selection tag

Commit:
GDEV-201 fixing selection_tags variable

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+
+# AWS Backup vault
+resource "aws_backup_vault" "vault" {
+  name        = join("-", [local.module_prefix, "vault"])
```
- [x] Add count attribute

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+
+# AWS Backup plan
+resource "aws_backup_plan" "plan" {
+  name = join("-", [local.module_prefix, "plan"])
```
- [x] Add count attribute

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+
+# AWS Backup selection - resource arn
+resource "aws_backup_selection" "arn_resource_selection" {
+  iam_role_arn = aws_iam_role.aws_backup_role.arn
```
- [x] Add count attribute

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+# KMS key for fsx
+
+resource "aws_kms_key" "default" {
+  count                    = var.create ? 1 : 0
```
- [x] Make count dynamic so kms key can be passed in

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+resource "aws_backup_plan" "plan" {
+  name = join("-", [local.module_prefix, "plan"])
+
+  rule {
```
- [x] Create dynamic rule with map instead of predefined rules

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+
+
+# Backup rules
+variable "daily_cron" {
```
- [x] Remove extra variables when modified for dynamic rule

TysonBeck commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,225 @@
+  default = []
+}
+
+
```
- [x] Add variable for passing in KMS key arn

Commit:
GDEV-201 replacing variables with dynamic blocks

Commit:
GDEV-201 fixing undeclared resources

Commit:
GDEV-201 adding count attribute

Commit:
GDEV-201 fixing Missing resource instance key

Commit:
GDEV-201 fixing rule argument

Commit:
GDEV-201 adding missed bracket

Commit:
GDEV-201 fixing target vault name

Commit:
GDEV-201 fixing IAM Policies

Commit:
GDEV-201 adding IAM variable

Commit:
GDEV-201 adding IAM variable and count

jeremyodle reviewed (changes requested):
- [x] Add `aws-backup` to .github Labeler
- [x] Update README

jeremyodle commented on aws/aws-backup/aws_backup.tf:
```diff
@@ -0,0 +1,327 @@
+
+resource "aws_kms_alias" "default" {
+  count         = var.create ? 1 : 0
+  name          = "alias/${replace(local.stage_prefix, var.delimiter, "/")}/aws-backup"
```
- [x] Use module name instead of hardcoding

TysonBeck committed:
GDEV-201 Added ability to pass in kms key arn

formatted
updated count logic
updated naming
removed workflow linter reporter as causing unnecessary spam
updated labeler
updated readme

TysonBeck committed:
GDEV-201 Fixing vault name logic

TysonBeck committed:
GDEV-201 Correcting syntax issue

jeremyodle reviewed (changes requested):
- [x] Rename module to `backup`

TysonBeck committed:
GDEV-201 Updating module name and dependent locations
